### PR TITLE
DNN: add another two Mish activation to onnx_graph_simplifier

### PR DESCRIPTION
**Pull request by zihaomu**
## Speed test on YoloV4

| Test in YoloV4 with M1 chip |  |  |
| --------------------------- | -- | -- |
|                             | Befor Patch optimize | With Patch optimize |
| Time                        | 364 ms | **304 ms (17% faster)**  |

The Mish layer can be generated by two different implementations:
```py
# PyTorch code 1: (We have supported in OpenCV.)
x * (torch.tanh(F.softplus(x)))

# PyTorch code 2: (Proposed: We do not support.)
x * (torch.tanh(torch.log(torch.exp(x) + 1)))
```

In order to make [YoloV4](https://github.com/onnx/models/tree/main/vision/object_detection_segmentation/yolov4) run faster on OpenCV, Mish's graph fusion optimization is essential.

For now, the original [YoloV4 in ONNX's model_zoo](https://github.com/onnx/models/tree/main/vision/object_detection_segmentation/yolov4) cannot be loaded correctly by OpenCV, because original YoloV4 has very complicated struct and the input is dynamic shape.

So I convert it to a suitable format by following python code:
```py
import onnx
import onnxsim
import numpy as np

input_model = "yolov4.onnx"
output_model = "yolov4_sim.onnx"
onnx_model = onnx.load(input_model)
input_shape = np.array([1, 416, 416, 3])
input = dict()
input["input_1:0"] = input_shape
out = onnxsim.simplify(onnx_model, dynamic_input_shape=False, input_shapes=input)
onnx.save(out[0], output_model)
```
And the converted model can be found at this [google drive](https://drive.google.com/file/d/1WBZO-YCqYTO73j8w_soMPD3ljp8fAD_-/view?usp=sharing).

[Test case in OpenCV_extra](https://github.com/opencv/opencv_extra/pull/990).

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/dnn/src/onnx/onnx_graph_simplifier.cpp [hunk 1]
```diff
@@ -531,6 +531,35 @@ class MishSubgraph : public Subgraph
...
+    SoftplusSubgraph2()
+    {
+        int input = addNodeToMatch("");
+        int exp = addNodeToMatch("Exp", input);
+        int addVal = addNodeToMatch("");
+        int add = addNodeToMatch("Add", exp, addVal);
+        addNodeToMatch("Log", add);
+        setFusedNode("Softplus", input);
+    }
+};
```
Comment (zihaomu) [inline modules/dnn/src/onnx/onnx_graph_simplifier.cpp:561]: We have two very similar `SoftplusSubgrap` because `Softplus` contains an `Add` which has two inputs (one is variable and one is `Constant` 1). Since the order of these two inputs is random, we use two `SoftplusSubgraph` to handle this. Is there a better solution?

Comment (rogday) [reply modules/dnn/src/onnx/onnx_graph_simplifier.cpp:561]: I think that simplifier should just match different nodes. You can make a small onnx model of both cases, remove the second class and test if it's working.

Comment (zihaomu) [reply modules/dnn/src/onnx/onnx_graph_simplifier.cpp:561]: Thanks for code reviewing! Actually, I have verified it. And the `SoftplusSubgraph` can work on [Yolov4](https://drive.google.com/file/d/1WBZO-YCqYTO73j8w_soMPD3ljp8fAD_-/view?usp=sharing), and the `SoftplusSubgraph2` can only work on [this test case](https://github.com/opencv/opencv_extra/pull/990).

**Review by vpisarev — APPROVED** [review]

File: modules/dnn/src/onnx/onnx_graph_simplifier.cpp (hunk 1 above)
Comment (rogday) [reply modules/dnn/src/onnx/onnx_graph_simplifier.cpp:561]: Hm, actually it makes sense - the operation is not required to be commutative. You could use CRTP to statically parametrise the order, but I don't think the complexity is worth it in this case. The second solution is to add logic of reordering into subgraph matching, but it'll take a while. I think this works for now.

**Review by rogday — APPROVED** [review]
LGTM :+1:
